milli/src/heed_codec/script_language_codec.rs

```diff
@@ -0,0 +1,43 @@
+use std::borrow::Cow;
+
+use std::mem::size_of;
+use std::str;
+
+use charabia::{Language, Script};
+
+pub struct ScriptLanguageCodec;
+
+impl<'a> heed::BytesDecode<'a> for ScriptLanguageCodec {
+    type DItem = (Script, Language);
+
+    fn bytes_decode(bytes: &'a [u8]) -> Option<Self::DItem> {
+        let footer_len = size_of::<u32>();
```

Comment:
Mmmmh, I'd prefer that this CODEC behave like the [StrStrU8Codec](https://github.com/meilisearch/milli/blob/fad0de4581248cdf748e3a7c5636310e0b21c0f8/milli/src/heed_codec/str_str_u8_codec.rs#L26-L29) where we encode `\0` to split the word instead of relying on the expected length of the Language code name.
Moreover, the size of an u32 has nothing to do with the language code name, so derivating the size from it is really obscure.